### ISSUE TITLE
webhook is not triggered properly

### ISSUE THREAD

**Issue by MehSha** [gogits/gogs#3977]
we use drone as CI, it was building projects on push and tag on gogs v 0.9.71 and when event was tag deploying them.
updated to 0.9.113.1223 and tag does not trigger ci to build. push works by chance, sometimes it gets 404 on delivery and sometimes works. same repository and CI works fine with push and tag on 0.9.71 with no single problem.
I tried tag, both through command line and release tab of project, pull request, etc. none trigger it. only push triggers which sometimes fails. also tried (I need EVERYTHING) with no luck.

- Gogs version: 0.9.113.1223 
- Git version: 1.8.3
- Operating system: linux ubuntu 16.04
- Database (use `[x]`):
  - [X ] PostgreSQL
  - [ ] MySQL
  - [ ] SQLite

**Comment by Unknwon** [comment]
Can you reproduce on https://try.gogs.io?

**Comment by Unknwon** [comment]
Push tag from command line 100% works, and for triggering webhook for a new release on web UI, this is duplicated to https://github.com/gogits/gogs/issues/2387.

**Comment by MehSha** [comment]
100 percent does not work, I wonder you close issue even without investigating. from 0.9.71 all your releases have been broken for CI use in some way, 0.9.97 does not allow CI to pull code, and this one does not trigger on push tag from command line.
to clear the matter, i do the following:
git tag -am "1.0" 1.0
git push origin 1.0
but NOTHING HAPPEN on CI.
one note: I first installed the 0.9.71 and databse was first create on that version. from then I only do binary update, just update and copy data/ log/ custom/ folder to new installation with same database as before. could it be something is wrong with binary update? is data base schema the same as before?
I have noticed you added webhook trigger on pull request. did that changes database schema in some way that I need to follow? or do fresh installation?
